### ISSUE TITLE
Make "Rasterize Categorized Vector Layer" cancelable

### ISSUE THREAD

**Issue by jakimowb** [EnMAP-Box/enmap-box#221]
Version: EnMAP-Box 3.11

It is not possible to cancel the "Raster Categorized Vector Layer" algorithm. 

![grafik](https://user-images.githubusercontent.com/1404870/197400291-c9602796-d4e3-48fe-9c08-23fdda640365.png)

**Comment by janzandr** [comment]
This is true for most of the algorithms. Let's make it a big issue, overhauling all the algorithms in that regard.
